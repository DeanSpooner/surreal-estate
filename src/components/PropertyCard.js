import React from "react";
import PropTypes from "prop-types";
import "../styles/PropertyCard.css";
import bath from "../images/bath.png";
import bed from "../images/bed.png";
import pound from "../images/pound.png";
import emailpic from "../images/email.png";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  return (
    <div className="propertyCard">
      <h2 className="propertyCard__title" data-testid="propertyCard__title">
        {title} in {city}
      </h2>
      <ul className="propertyCard__list">
        <li className="propertyCard__type">
          {type} in {city}
        </li>
        <li className="propertyCard__bathrooms">
          <img src={bath} alt="Bath icon" className="bath" /> {bathrooms}{" "}
          {bathrooms === 1 ? `bathroom` : `bathrooms`}
        </li>
        <li className="propertyCard__bedrooms">
          <img src={bed} alt="Bed icon" className="bed" /> {bedrooms}{" "}
          {bedrooms === 1 ? `bedroom` : `bedrooms`}
        </li>
        <li className="propertyCard__price">
          <img src={pound} alt="Pound icon" className="pound" /> Â£{price}
        </li>
      </ul>
      <li className="propertyCard__email">
        <a href={`mailto:${email}`} className="propertyCard__emailhref">
          <img src={emailpic} alt="Email icon" className="email" /> Click to
          email seller
        </a>
      </li>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
