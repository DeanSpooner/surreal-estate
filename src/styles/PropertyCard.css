.propertyCard {
  background-color: #b3a5d0;
  color: #000;
  display: inline-block;
  margin: 1em;
  width: 45%;
  border-radius: 12px;
}

h2,
li {
  list-style-type: none;
  margin-bottom: 1em;
  margin-left: 1em;
}

.bath,
.bed,
.pound {
  position: relative;
  top: 0.2em;
  width: 1.5em;
}

.email {
  position: relative;
  top: 0.2em;
  width: 1em;
}

.propertyCard__title {
  padding-top: 1em;
}

.propertyCard__email {
  background-color: rgb(131, 21, 199);
  border-radius: 12px 12px 0 0;
  font-size: 2em;
  margin-left: 1em;
  margin-right: 1em;
  padding-bottom: 1em;
  text-align: center;
  width: (100% - 2em);
  position: relative;
  top: 1em;
}

.propertyCard__email > a {
  position: relative;
  top: 0.5em;
}
