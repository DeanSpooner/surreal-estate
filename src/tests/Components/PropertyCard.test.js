import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "2 bed flat",
    type: "Flat",
    bathrooms: 1,
    bedrooms: 2,
    price: 125000,
    city: "Liverpool",
    email: "dean.spooner@here.com",
  };

  const { asFragment } = render(<PropertyCard props={validProps} />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<PropertyCard props={validProps} />);

    expect(getByText("2 bed flat in Liverpool")).toHaveClass(
      "propertyCard__title"
    );
    expect(getByText("Flat in Liverpool")).toHaveClass("propertyCard__type");
    expect(getByText("1 bathroom")).toHaveClass("propertyCard__bathrooms");
    expect(getByText("2 bedrooms")).toHaveClass("propertyCard__bedrooms");
    expect(getByText("Â£125000")).toHaveClass("propertyCard__price");
    expect(getByText("Click to email seller")).toHaveClass(
      "propertyCard__emailhref"
    );
  });
});
