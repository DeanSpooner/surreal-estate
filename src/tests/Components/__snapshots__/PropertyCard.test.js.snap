// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="propertyCard"
  >
    <h2
      class="propertyCard__title"
      data-testid="propertyCard__title"
    >
      2 bed flat in Liverpool
    </h2>
    <ul
      class="propertyCard__list"
    >
      <li
        class="propertyCard__type"
      >
        Flat in Liverpool
      </li>
      <li
        class="propertyCard__bathrooms"
      >
        <img
          alt="Bath icon"
          class="bath"
          src="bath.png"
        />
         1 bathroom
      </li>
      <li
        class="propertyCard__bedrooms"
      >
        <img
          alt="Bed icon"
          class="bed"
          src="bed.png"
        />
         2 bedrooms
      </li>
      <li
        class="propertyCard__price"
      >
        <img
          alt="Pound icon"
          class="pound"
          src="pound.png"
        />
         Â£125000
      </li>
    </ul>
    <li
      class="propertyCard__email"
    >
      <a
        class="propertyCard__emailhref"
        href="mailto:dean.spooner@here.com"
      >
        <img
          alt="Email icon"
          class="email"
          src="email.png"
        />
         Click to email seller
      </a>
    </li>
  </div>
</DocumentFragment>
`;
